src/lib.rs

```diff
@@ -0,0 +1,114 @@
+//! Interfaces for sequences.
+//!
+//! This crate defines the traits [`Sequence`] and [`SequenceMut`] providing a
+//! subset of the methods defined for [`slice`]. The reason for existence is
+//! that [`Sequence`] and [`SequenceMut`] can be implemented for types other
+//! than those that can (unconditionally) be borrowed as a slice, e.g.
+//! [`ndarray::Array1`].
+//!
+//! The [`Sequence`] and [`SequenceMut`] traits are implemented for the
+//! following types:
+//!
+//! *   [`slice`]
+//! *   [`array`]
+//! *   [`std::collections::VecDeque`] (requires feature `alloc`),
+//! *   [`ndarray::Array1`] (requires feature `ndarray`),
+//!
+//! There are deliberately no implementations for types like
+//! [`Vec`][`std::vec::Vec`] and [`Box<T>`][`std::boxed::Box`], where `T`
+//! implements [`Sequence`], in favor of automatic dereferencing to a
+//! [`slice`] and `T`, respectively, or the dereference traits [`AsSequence`]
+//! and [`AsSequenceMut`] detailed below.
```

Comment:
AsSqnc and AsMutSqnc. I guess details still need to be written. Also, is it _or_ the dereference traits, or _via_? Otherwise I think "automatic" should be "manual".